src/solvers/petsc_linear_solver.C

```diff
@@ -158,17 +140,22 @@ void PetscLinearSolver<T>::init (const char * name)
       PetscErrorCode ierr=0;
 
       // Create the linear solver context
-      ierr = KSPCreate (this->comm().get(), &_ksp);
-      LIBMESH_CHKERR(ierr);
+      _ksp = [&]()
+        {
+          KSP tmp;
+          ierr = KSPCreate (this->comm().get(), &tmp);
+          LIBMESH_CHKERR(ierr);
+          return std::unique_ptr<KSP, KSPDeleter>(&tmp);
+        }();
```

Comment:
This just doesn't work at all. I kind of thought that `tmp` here is like a heap-allocated pointer for some reason... but that seems to not be the case. In the debugger, I see:
```
(gdb) p _ksp
$1 = std::unique_ptr<_p_KSP *> = {get() = 0x7ffdd3e93ec8}
```
but
```
(gdb) p *_ksp
$2 = (_p_KSP *) 0x0
```
I still think there is an improvement possible that would allow us to avoid making manual `XXXDestroy()` calls, but this ain't it.